rr_iarrc/src/stopbar_detector/stopbar_detector.cpp

```diff
@@ -0,0 +1,186 @@
+
+                // draw line to stopbar in debug image and displays the distance in meters to it
+                cv::line(output, midpoint, cv::Point(midpoint.x, edges.rows), cv::Scalar(0, 255, 255), 1, cv::LINE_AA);
+                std::stringstream streamDist;
+                streamDist << std::fixed << std::setprecision(2) << dist;  // show distance with a couple decimals
+                cv::putText(output, streamDist.str(), cv::Point(midpoint.x, edges.rows - dist / 2),
+                            cv::FONT_HERSHEY_PLAIN, 1, cv::Scalar(0, 255, 0), 1);
+
+                stopBarAngle = currAngle;
+                if (speed != 0) {
+                    double timeToStopBar = (dist / speed) * sleepConstant;
+                    ros::Duration(timeToStopBar).sleep();
+                }
+                return true;  // stop bar detected close to us!
+            }
+        }
+    }
+
+    return false;  // not close enough or no stop bar here
+}
+
+void stopBar_callback(const sensor_msgs::ImageConstPtr& msg) {
+    cv_ptrLine = cv_bridge::toCvCopy(msg, "mono8");
+    cv::Mat frame = cv_ptrLine->image;
+    cv::Mat debug;
+    double stopBarAngle;
+    bool stopBarDetected =
+          findStopBarFromHough(frame, debug, stopBarAngle, stopBarGoalAngle, stopBarGoalAngleRange,
+                               stopBarTriggerDistance, houghThreshold, houghMinLineLength, houghMaxLineGap);
+
+    // debugging draw a line where we trigger
+    cv::Point leftPoint(0, debug.rows - 1 - stopBarTriggerDistance * pixels_per_meter);
+    cv::Point rightPoint(debug.cols - 1, debug.rows - 1 - stopBarTriggerDistance * pixels_per_meter);
+    cv::line(debug, leftPoint, rightPoint, cv::Scalar(0, 255, 0), 1, cv::LINE_AA);
+
+    stop_bar_near.data = stopBarDetected;
+    if (stopBarDetected)
+        stop_bar_angle.data = stopBarAngle;
+    pub_near_stopbar.publish(stop_bar_near);
+    pub_angle.publish(stop_bar_angle);
+    if (pub_line.getNumSubscribers() > 0) {
+        sensor_msgs::Image outmsg;
+        cv_ptrLine->image = debug;
+        cv_ptrLine->encoding = "bgr8";
+        cv_ptrLine->toImageMsg(outmsg);
+        pub_line.publish(outmsg);
+    }
+}
+
+void speed_callback(const rr_msgs::speed& speedMsg) {
+    speed = speedMsg.speed;
+}
+
+int main(int argc, char** argv) {
+    ros::init(argc, argv, "sign_detector");
+
+    ros::NodeHandle nh;
+    ros::NodeHandle nhp("~");
+
+    std::string overhead_image_sub;
+    std::string stopbar_near_topic;
+    std::string stopbar_angle_topic;
+    std::string speed_topic;
+
+    nhp.param("stopbar_near", stopbar_near_topic, std::string("/stopbar_near"));
+    nhp.param("stopbar_angle", stopbar_angle_topic, std::string("/stopbar_angle"));
+    nhp.param("sleep_adjust", sleepConstant, 1.0);
+
+    nhp.param("overhead_image_subscription", overhead_image_sub, std::string("/lines/detection_img_transformed"));
+
+    nhp.param("speed_subscription", speed_topic, std::string("/speed"));
```

Comment:
If you are going to do slash + topic name (`/blah`), you can just do `nh.param`, but you don't have to change it